### ISSUE TITLE
AlignArguments (and AlignParameters probably) interaction with AlignHash w/ignore_implicit

### ISSUE THREAD

**Issue by lamont-granquist** [rubocop/rubocop#7198]
with these settings:

```
Layout/AlignArguments:
  Enabled: true
  EnforcedStyle: with_fixed_indentation
Layout/AlignParameters:
  Enabled: true
  EnforcedStyle: with_fixed_indentation
Layout/AlignHash:
  Enabled: true
  EnforcedLastArgumentHashStyle: ignore_implicit
```

I wouldn't expect this to be legal:

```ruby
def rest
  @rest ||= Chef::ServerAPI.new(
    Chef::Config[:chef_server_url], client_name: node_name,
                                signing_key_filename: Chef::Config[:client_key]
  )
end
```

Because the kwargs there are an implicit hash they're being ignored by AlignHash correctly, but what I'd like is for AlignArguments to pick up the slack and fix it to:

```ruby
def rest
  @rest ||= Chef::ServerAPI.new(
    Chef::Config[:chef_server_url], client_name: node_name,
    signing_key_filename: Chef::Config[:client_key]
  )
end
```

The first snippet in a .rubocop.yml with all the rest of the default rubocop rules passes on either version of this code so there's no enforcement at all of indentation on that line with these settings, and apparently no other default cop setting that would pick it up.

**Comment by allentsai** [comment]
Same with us.
config:
```
Layout/AlignArguments:
  Enabled: true

Layout/AlignHash:
  Enabled: true
  EnforcedLastArgumentHashStyle: ignore_implicit

Layout/AlignParameters:
  Enabled: true
  EnforcedStyle: with_fixed_indentation
```
valid:
```
      EmailSubscriptionFeedEntry.create!(
        feed_sync_id: max_feed_sync_id + processed,
          user_token: email_user_opt_in.user_token,
        email_user_opt_in_token: email_user_opt_in.token
      )
```
But we would like the `user_token` key to be aligned.
It might not be that simple though--we introduced the `EnforcedLastArgumentHashStyle: ignore_implicit` because something like:
```
render key1: arg1
  key2: arg2
```
was not being handled the way we wanted.

**Comment by mvz** [comment]
This problem still occurs with rubocop 0.80.0.

**Comment by lamont-granquist** [comment]
validated in 0.86.0 still

**Comment by Earlopain** [comment]
This can be closed, I tried this with current RuboCop (1.69.2) and somewhere in the last 4 years it now behaves as expected.
